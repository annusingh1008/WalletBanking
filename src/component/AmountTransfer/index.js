import React, { useState } from 'react'
import Layout from '../Layout'
import { Row, Col, Container, Button, Form, Modal } from 'react-bootstrap'
import { useDispatch, useSelector } from 'react-redux';
import  {amountTransfer}  from '../../actions/amountTransfer.action';
import { getUserDetails } from '../../actions/getUserDetails.actions';

const AmountTransfer = () => {

  const [confirmModal, setConfirmModal] = useState(false);
  const [amount, setAmount] = useState();
  const [email, setEmail] = useState('');

  const dispatch = useDispatch();
  const auth = useSelector(state => state.auth);
  const balance = useSelector(state => state.balance);

  // const confirmTransaction = (e) => {
  //   e.preventDefault();
  //   setConfirmModal(true);
  // }

  const confirmAmountTransfer = () => {
      console.log("confirm amount transfer");
      const userEmail = localStorage.getItem('email');
      const amountDetails = {
        email: userEmail,
        amount: amount, 
        creditToEmail: email,
        currentAmount: balance.amount,
      }
      
      dispatch(amountTransfer(amountDetails))
  
  }

  return (
    <Layout sidebar>
      <Container className='container'>
        <Row md={12}>
          <Col>
            <div className='heading'>Transfer Amount</div>
            <Form className='form' onSubmit={confirmAmountTransfer} >
              <label>Email</label><br />

              <input
                type="email"
                placeholder='Email'
                value={email}
                onChange={(e) => setEmail(e.target.value)}
                required
              ></input><br />

              <label>Amount</label><br />

              <input
                type="text"
                placeholder='1000'
                value={amount}
                onChange={(e) => setAmount(e.target.value)}
                required
              ></input><br />

              <Button type='variant'>Transfer Amount</Button>
            </Form>
          </Col>
        </Row>

        {/* {confirmModal && (
          <Modal
            show={confirmModal}
            onHide={() => {setConfirmModal(false)}}
            backdrop="static"
            keyboard={false}
          >
            <Modal.Header closeButton>
              <Modal.Title>Confirm</Modal.Title>
            </Modal.Header>
            <Modal.Body>
              Are you sure, you want to transfer Rs.{amount} to Jack
            </Modal.Body>
            <Modal.Footer>
              <Button
                variant="secondary"
                onClick={() => setConfirmModal(false)}
              >
                Close
              </Button>
              <Button variant="primary" onClick={confirmAmountTransfer} >Confirm</Button>
            </Modal.Footer>
          </Modal>
        )} */}
      </Container>
    </Layout>
  )
}

export default AmountTransfer
