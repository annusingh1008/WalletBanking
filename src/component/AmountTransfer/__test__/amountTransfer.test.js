import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import AmountTransfer from "../index";
import { Provider } from "react-redux";
import store from "../../../store/index";
import { BrowserRouter as Router } from "react-router-dom";
import { getBalance } from "../../../reducers/balance.reducer";
import { Container } from "react-bootstrap";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch,
}));

jest.mock("../../../actions/amountTransfer.action", () => ({
  AmountTransfer: jest.fn(() => ({ type: "amountTransfer" })),
}));

jest.mock("../../../reducers/balance.reducer", () => ({
  getBalance: jest.fn(() => ({ amount: 1000 })),
}));

jest.mock("../../Layout", () =>
  jest.fn(({ children }) => <div data-testid="layout">{children}</div>)
);

jest.mock("react-bootstrap", () => ({
  Row: jest.fn(({ children }) => <div data-tesid="row">{children}</div>),
  Col: jest.fn(({ children }) => <div data-tesid="col">{children}</div>),
  Container: jest.fn(({ children }) => (
    <div data-tesid="container">{children}</div>
  )),
  Button: jest.fn(({ children }) => <div data-tesid="button">{children}</div>),
  Form: jest.fn(({ children }) => <div data-tesid="form">{children}</div>),
}));

const localStorageData = {
  email: "annu",
  password: "1234",
};

const localStorageMock = (function () {
  return {
    getItem: function (key) {
      return localStorageData[key];
    },
    removeItem: function () {
      return null;
    },
    setItem: jest.fn(),
  };
})();

describe("AmountTransfer", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  Object.defineProperty(window, "localStorage", { value: localStorageMock });

  // it("should render amount and email input element", () => {
  //   render(
  //     <Provider store={store}>
  //       <Router>
  //         <AmountTransfer />
  //       </Router>
  //     </Provider>
  //   );

  //   const inputElement1 = screen.getByPlaceholderText(/Email/i);
  //   expect(inputElement1).toBeInTheDocument();
  //   const inputElement2 = screen.getByPlaceholderText(/1000/i);
  //   expect(inputElement2).toBeInTheDocument();
  // });

  // it("should render email and amount label element", () => {
  //   render(
  //     <Provider store={store}>
  //       <Router>
  //         <AmountTransfer />
  //       </Router>
  //     </Provider>
  //   );

  //   const labeElement1 = screen.getByText("Email");
  //   expect(labeElement1).toBeInTheDocument();
  //   const labeElement2 = screen.getByText("Amount");
  //   expect(labeElement2).toBeInTheDocument();
  // });

  // it("should render Amount Transfer button", () => {
  //   render(
  //     <Provider store={store}>
  //       <Router>
  //         <AmountTransfer />
  //       </Router>
  //     </Provider>
  //   );

  //   const buttonElement = screen.getByText("Amount Transfer");
  //   expect(buttonElement).toBeInTheDocument();
  // });

  // it("renders properly", () => {
  //   const { container } = render(
  //     <Provider store={store}>
  //       <Router>
  //         <AmountTransfer />
  //       </Router>
  //     </Provider>
  //   );
  //   expect(container.firstChild).toMatchSnapshot();
  // });

  // it("should be able to type amount transfer details", () => {
  //   render(
  //     <Provider store={store}>
  //       <Router>
  //         <AmountTransfer />
  //       </Router>
  //     </Provider>
  //   );

  //   const textAreaElement1 = screen.getByPlaceholderText(/Email/i);
  //   fireEvent.change(textAreaElement1, { target: { value: "annu@gmail.com" } });
  //   expect(textAreaElement1.value).toBe("annu@gmail.com");

  //   const textAreaElement2 = screen.getByPlaceholderText(/1000/i);
  //   fireEvent.change(textAreaElement2, { target: { value: "1000" } });
  //   expect(textAreaElement2.value).toBe("1000");
  // });

  // describe("#getItem", () => {
  //   it("returns the user email", () => {
  //     const result = localStorage.getItem("email");
  //     expect(result).toEqual("annu");
  //   });
  // });

  // it("Form can be submitted", () => {
  //   const mockSubmit = jest.fn();
  //   const { queryByTestId } = render(
  //     <Provider store={store}>
  //       <Router>
  //         <AmountTransfer />
  //       </Router>
  //     </Provider>
  //   );
  //   fireEvent.submit(queryByTestId("button"));
  //   fireEvent.submit(queryByTestId("form"));
  //   expect(mockDispatch).toHaveBeenCalled();
  // });

  describe("when user puts insufficient amount", () => {
    it("Shows insufficient amount", () => {
      const { container } = render(
        <Provider store={store}>
          <Router>
            <AmountTransfer />
          </Router>
        </Provider>
      );
      console.log("store", store);
      const emailElement = screen.queryByTestId("emailInputBox");
      const amountElement = screen.queryByTestId("amountInputBox");
      fireEvent.change(emailElement, { target: { value: "annu@gmail.com" } });
      fireEvent.change(amountElement, { target: { value: 20000 } });
      // expect(screen.queryByTestId("insufficient-amount")).toBeVisible();
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
