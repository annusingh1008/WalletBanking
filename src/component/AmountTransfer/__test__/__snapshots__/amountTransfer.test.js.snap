// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountTransfer when user puts insufficient amount Shows insufficient amount 1`] = `
<div
  data-testid="layout"
>
  <div
    data-tesid="container"
  >
    <div
      data-tesid="row"
    >
      <div
        data-tesid="col"
      >
        <div
          class="heading"
        >
          Transfer Amount
        </div>
        <div
          data-tesid="form"
        >
          <label>
            Email
          </label>
          <br />
          <input
            data-testid="emailInputBox"
            placeholder="Email"
            required=""
            type="email"
            value="annu@gmail.com"
          />
          <br />
          <label>
            Amount
          </label>
          <br />
          <input
            data-testid="amountInputBox"
            placeholder="1000"
            required=""
            type="text"
            value="20000"
          />
          <br />
          <div
            data-tesid="button"
          >
            Transfer Amount
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
